functions/functionsv2/helloworld/hello_storage.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START functions_cloudevent_storage]
+
+// Package helloworld provides a set of Cloud Functions samples.
+package helloworld
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+)
+
+// StorageObjectData contains metadata of the Cloud Storage object.
+type StorageObjectData struct {
+	Bucket         string    `json:"bucket,omitempty"`
+	Name           string    `json:"name,omitempty"`
+	Metageneration int64     `json:"metageneration,string,omitempty"`
+	TimeCreated    time.Time `json:"timeCreated,omitempty"`
+	Updated        time.Time `json:"updated,omitempty"`
+}
+
+// HelloStorage consumes a CloudEvent message and logs details about the changed object.
+func HelloStorage(ctx context.Context, e event.Event) error {
```

Comment:
Did we want to split each function into a separate folder?

Reply:
Sure, I copied the existing model. We can go sample per directory.

Reply:
What is the benefit of a separate folder?

Reply:
Having a separate folder is a spanning decision across all languages for GCF v2 samples. For more discussion, please engage with @ace-n. Since this is not the PR which creates the new directory, I'd prefer to proceed as-is.

Reply:
@grayside post-facto split seems fine, but please tell TW (@labtopia) to hold off on including this sample **until the splitting PR is merged.**

@tbpg one of the primary benefits of separate folders is avoiding dependency pollution: each function can have its own dependencies without affecting another's.

Reply:
TW informed.